ward/suite.py

```diff
@@ -34,31 +34,13 @@ def generate_test_runs(self, order="standard") -> Generator[TestResult, None, No
             generated_tests = test.get_parameterised_instances()
             num_tests_per_module[test.path] -= 1
             for i, generated_test in enumerate(generated_tests):
-                marker = generated_test.marker.name if generated_test.marker else None
-                if marker == "SKIP":
-                    yield generated_test.get_result(TestOutcome.SKIP)
-                    continue
-
-                try:
-                    resolved_vals = generated_test.resolve_args(self.cache, iteration=i)
-                    generated_test.format_description(resolved_vals)
-                    generated_test(**resolved_vals)
-                    outcome = (
-                        TestOutcome.XPASS if marker == "XFAIL" else TestOutcome.PASS
-                    )
-                    yield generated_test.get_result(outcome)
-                except FixtureError as e:
-                    yield generated_test.get_result(TestOutcome.FAIL, e)
-                    continue
-                except Exception as e:
-                    outcome = (
-                        TestOutcome.XFAIL if marker == "XFAIL" else TestOutcome.FAIL
-                    )
-                    yield generated_test.get_result(outcome, e)
-                finally:
-                    self.cache.teardown_fixtures_for_scope(
-                        Scope.Test, scope_key=generated_test.id
-                    )
+                resolved_vals = generated_test.resolve_args(self.cache, iteration=i)
```

Comment:
I need to spend more time grokking the fixture cache and resolution in order to give you an informed response to this. However, I will say that I would think that fixture cache handling would still be the prerogative of the Suite class rather than the Test class, since fixtures can last for longer than one test.

Reply:
> However, I will say that I would think that fixture cache handling would still be the prerogative of the Suite class rather than the Test class, since fixtures can last for longer than one test.

Makes sense, I understand your concerns. 

Here's some more background on how fixture caching works:

The `FixtureCache` has 3 layers and a Test with generated ID `"test-id"` would only access the namespace `cache[Scope.Test]["test-id"]` within the cache. If that test took 2 test-scoped fixtures, `fixture_a` and `fixture_b`, the `fixture_cache` might have keys:

```
fixture_cache[Scope.Test]["test-id"]["path/to/fixtures.py::fixture_a"]
fixture_cache[Scope.Test]["test-id"]["path/to/more_fixtures.py::fixture_b"]
```

* So `cache[Scope.Test]["test-id"]` contains all fixtures that need to be cleaned up after test with ID `"test-id"` completes. In otherwords, all fixtures that are alive for the duration of the test.
* `cache[Scope.Module][Path("tests/test_x.py")]` contains all the fixtures that need to be cleaned up after the module at path `tests/test_x.py` completes. They're cached from the first time they're required in `test_x`, until the final test in `test_x` completes.
* `cache[Scope.Global][Scope.Global]` contains all the fixtures that need to be cleaned up after all tests complete. They're cached from the first time they're required in a session, and cleaned up after all collected tests have complete.

Hopefully that helps!

Reply:
Yes, this helped thanks :)